pkg/endpoint/builder.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"github.com/cilium/cilium/pkg/buildqueue"
+	"github.com/cilium/cilium/pkg/monitor"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+var (
+	// BuildQueue is the endpoint build queue. It is also used to build
+	// base programs
+	BuildQueue = buildqueue.NewBuildQueue("endpoint-builder")
+)
+
+type endpointBuild struct {
+	endpoint *Endpoint
+	context  *RegenerationContext
+	owner    Owner
+}
+
+func (e *Endpoint) newEndpointBuild(owner Owner, context *RegenerationContext) *endpointBuild {
+	return &endpointBuild{
+		endpoint: e,
+		owner:    owner,
+		context:  context,
+	}
+}
+
+func (b *endpointBuild) GetUUID() string {
+	return b.endpoint.GetUUID()
```

Comment:
The UUID does not describe the build. It describes the buildable. If the same buildable is being built multiple times, all of these builds will have the same UUID and are correlated as such. We previously used the endpoint.ID for this purpose but the endpoint.ID is fragile as it can be reused again. Introducing a UUID allows grepping for the UUID in the logs.